secure/benches/benchmarks.rs

```diff
@@ -187,6 +191,47 @@ fn benchmark_handshake(c: &mut Criterion) {
     });
 }
 
+fn benchmark_symmetric_encryption(c: &mut Criterion) {
+    c.bench_function(
+        "benchmark_symmetric_encryption_data_size_16008_bytes",
+        |b| {
+            let mut rng = rand::thread_rng();
+            let mut message: Vec<u8> = vec![0; 16 * 100 + 8];
+            rng.fill_bytes(&mut message);
+            let mut key = [0; 16];
+            rng.fill_bytes(&mut key);
+            let security = SymmetricSecurity::from(&key);
+            b.iter(|| {
+                let Ok((_cipher, _nonce, _padding)) = security.encrypt(&message) else {
+                    assert!(false);
+                    return;
+                };
+            })
+        },
+    );
+}
+
+fn benchmark_symmetric_decryption(c: &mut Criterion) {
+    c.bench_function(
+        "benchmark_symmetric_decryption_data_size_16008_bytes",
+        |b| {
+            let mut rng = rand::thread_rng();
+            let mut message: Vec<u8> = vec![0; 16 * 100 + 8];
+            rng.fill_bytes(&mut message);
+            let mut key = [0; 16];
+            rng.fill_bytes(&mut key);
+            let security = SymmetricSecurity::from(&key);
+            let Ok((cipher, nonce, padding)) = security.encrypt(&message) else {
+                assert!(false);
+                return;
+            };
+            b.iter(|| {
+                let _plane = security.decrypt(&cipher, &nonce, padding);
+            })
```

Comment:
Suggestion: I would remove a function for encryption.

Having function for decryption requires encryption too. So benchmark is for encryption-decryption, not decryption only.

Reply:
They are separate actions and can be altered in the future separately, so I want to have a benchmark measuring each one.

If you will in the future alter the decryption body and encryption body making the first faster and the second slower then: the execution time sum may be the same but individually you made an improvement in the first and the regression of the second. But in the compound tests, you cannot see it.

Reply:
So this one test is the latter

Reply:
Ok,now I got it. Look that `b.iter` loops over only `let _plane = security.decrypt(&cipher, &nonce, padding);` and this is what is benchmarked.